Adds CLI for Microsoft 365 10.5.0 release announcement (#2064)
pnp/blog: content/cli-for-microsoft-365/cli-for-microsoft-365-v10-5/index.md

---
title: CLI for Microsoft 365 v10.5
date: 2025-03-31T01:00:00.000Z
author: Martin Lingstuyl
githubname: martinlingstuyl
categories:
  - CLI for Microsoft 365
images:
  - images/banner-cli-m365.png
tags:
  - CLI for Microsoft 365
  - Viva Engage
  - SharePoint
  - Microsoft Graph
type: popular
---

It's that time of the month again! We've just published a new minor version of the CLI for Microsoft 365. [CLI for Microsoft 365](https://aka.ms/cli-m365) is a cross-platform command-line tool that allows you to manage your Microsoft 365 tenant and SharePoint Framework projects. This minor release introduces new commands that will enhance the quality of your Microsoft 365 experience.

> > Explore the [release notes](https://aka.ms/cli-m365/notes) for a neat overview of all the exciting features and improvements. 

Reviewer: I would add a link with an anchor to the v10.5 section.

Author: Forgot this one, well ok, not that important 😊

## What's New

### Microsoft Graph

> In this release, we added three new commands to manage directory extensions for Entra ID through the Microsoft Graph. Directory extensions are a way to add additional properties to Entra ID resources: users, groups, administrative units, applications, devices or organizations. They're a flexible way to enrich your company data with information that's relevant to you. Using the below new commands you can define new directory extensions, remove these definitions, and retrieve them.

Reviewer 2: ```suggestion
In this release, we added three new commands to manage directory extensions for Entra ID through the Microsoft Graph. Directory extensions are a way to add additional properties to Entra ID resources: users, groups, administrative units, applications, devices, or organizations. They're a flexible way to enrich your company data with information that's relevant to you. Using the below new commands you can define new directory extensions, remove these definitions, and retrieve them.
```

To define a new directory extension for saving GitHub information for Entra users, run:

```sh
m365 graph directoryextension add --name gitHubWorkAccountName --appName ContosoApp --targetObjects User --dataType String
```

To retrieve a specific directory extension, run:

```sh

> m365 directoryextension get --name extension_105be60b603845fea385e58772d9d630_GitHubWorkAccount --appName ContosoApp

Reviewer: ```suggestion
m365 graph directoryextension get --name extension_105be60b603845fea385e58772d9d630_GitHubWorkAccount --appName ContosoApp
```

```

To remove the directory extension, run:

```sh
m365 graph directoryextension remove --name extension_105be60b603845fea385e58772d9d630_GitHubWorkAccount --appName ContosoApp --force
```

For more information, check the following resources:

- [graph directoryextension add](https://pnp.github.io/cli-microsoft365/cmd/graph/directoryextension/directoryextension-add/)
- [graph directoryextension get](https://pnp.github.io/cli-microsoft365/cmd/graph/directoryextension/directoryextension-get/)
- [graph directoryextension remove](https://pnp.github.io/cli-microsoft365/cmd/graph/directoryextension/directoryextension-remove/)

### SharePoint

> This release, we're also adding two new commands for SharePoint. It's now possible to add new home sites, making it easier to create new Viva Connections experiences, each on a standalone home site. Or how about removing sections on a sharepoint page. It's now possible in our effort to make it easier to design SharePoint sites using scripting, for example in a migration scenario.

Reviewer 2: ```suggestion
In this release, we're also adding two new commands for SharePoint. It's now possible to add new home sites, making it easier to create new Viva Connections experiences, each on a standalone home site. Or how about removing sections on a SharePoint page. It's now possible in our effort to make it easier to design SharePoint sites using scripting, for example, in a migration scenario.
```

```

For more information, check the following resources:

- [m365 spo homesite add](https://pnp.github.io/cli-microsoft365/cmd/spo/homesite/homesite-add)
- [m365 spo page section remove](https://pnp.github.io/cli-microsoft365/cmd/spo/page/page-section-remove)

### Viva Engage 

Managing Viva Engage communities has also been enhanced with two new commands. It's now possible to add and remove community users.

To add a new user as an admin to a specific community, run:

```sh
m365 viva engage community user add --communityDisplayName "Innovation" --userNames john.doe@contoso.com --role Admin
```

To remove a user from a specific community, run:

```sh
m365 viva engage community user remove --communityDisplayName "Innovation" --userName john.doe@contoso.com --force
```

For more information, check the following resources:

- [m365 viva engage community user add](https://pnp.github.io/cli-microsoft365/cmd/viva/engage/engage-community-user-add)
- [m365 viva engage community user remove](https://pnp.github.io/cli-microsoft365/cmd/viva/engage/engage-community-user-remove)

## What's changed

> This release also brings some other great changes. Aside from several enhancements and bug fixes to elevate the overall CLI experience, there's a couple of things we'd like to highlight.

Reviewer 2: ```suggestion
This release also brings some other great changes. Aside from several enhancements and bug fixes to elevate the overall CLI experience, there are a couple of things we'd like to highlight.
```

### New sample script - Remove SharePoint page header

We've also added a new sample regarding page management. With this sample it's easier to automatically remove the page header from a page. 

For more information check the following resource:

[Remove SharePoint page header](https://pnp.github.io/cli-microsoft365/sample-scripts/spo/remove-page-header/)

### New guide - Using CLI in an Azure Function with PowerShell

Did you know you can run CLI for Microsoft 365 scripts in an Azure Function? It's a great way to automate your scripts with a tool that's familiar to you. You could write your script on your machine, testing it out as you go. As soon as you're ready, create a new function app in Azure and deploy your script to run every day or on whatever schedule or with whatever trigger you like. 

This guide helps to get you started! 

For more information check the following resource:

[Using CLI in an Azure Function with PowerShell](https://pnp.github.io/cli-microsoft365/user-guide/automation/azure-function-powershell/)

### Using Federated Identity in GitHub actions

> Deploying apps in GitHub workflows is getting easier! Drop all your certificates and implement Federated Credentials! Federated identity is quite recent addition to Entra ID, which means you can establish a trust relationship between a third-party service like GitHub and Entra ID. It means that you can start using credentialless authentication for the CLI for Microsoft 365! You'll be able to configuring an Entra application to allow GitHub Actions to sign in. Deploying an SPFx app using GitHub Actions becomes easier like this. You won't have to keep certificate expiration dates in mind and you don't need to be afraid that your certificate is compromised... No credentials, better security!

Reviewer 2: ```suggestion
Deploying apps in GitHub workflows is getting easier! Drop all your certificates and implement Federated Credentials! Federated identity is a quite recent addition to Entra ID, which means you can establish a trust relationship between a third-party service like GitHub and Entra ID. It means that you can start using credentialless authentication for the CLI for Microsoft 365! You'll be able to configure an Entra application to allow GitHub Actions to sign in. Deploying an SPFx app using GitHub Actions becomes easier like this. You won't have to keep certificate expiration dates in mind and you don't need to be afraid that your certificate is compromised... No credentials, better security!
```
